src/hooks/useCart/useCart.test.tsx

```diff
@@ -1203,6 +1203,147 @@ describe("useCart", () => {
     });
   });
 
+  describe("complete checkout", () => {
+    it("should set the correct checkoutResult when completeCheckout is called", async () => {
+      expect.assertions(2);
+      const ids = ["ID1"];
+      const { result } = renderHook(
+        () => useCart(ids, key, endpoint, mockQuotaResSingleId.remainingQuota),
+        {
+          wrapper,
+        }
+      );
+
+      await waitFor(() => {
+        result.current.updateCart("toilet-paper", 1, [
+          {
+            label: "second",
+            value: "second",
+            textInputType: "PAYMENT_RECEIPT",
+          },
+        ]);
+      });
+
+      mockPostTransaction.mockReturnValueOnce(mockPostTransactionResult);
+
+      await waitFor(() => {
+        result.current.checkoutCart();
+        expect(result.current.cartState).toBe("PENDING_CONFIRMATION");
+      });
+
+      await waitFor(() => {
+        result.current.completeCheckout();
+      });
+
+      expect(result.current.cart).toStrictEqual([
+        {
+          category: "toilet-paper",
+          descriptionAlert: undefined,
+          identifierInputs: [
+            {
+              label: "second",
+              textInputType: "PAYMENT_RECEIPT",
+              value: "second",
+            },
+          ],
+          lastTransactionTime: transactionTime,
+          maxQuantity: 2,
+          quantity: 1,
+        },
+        {
+          category: "chocolate",
+          descriptionAlert: undefined,
+          identifierInputs: [
+            {
+              label: "first",
+              scanButtonType: "BARCODE",
+              textInputType: "STRING",
+              value: "",
+            },
+            {
+              label: "last",
+              scanButtonType: "BARCODE",
+              textInputType: "STRING",
+              value: "",
+            },
+          ],
+          lastTransactionTime: transactionTime,
+          maxQuantity: 15,
+          quantity: 0,
+        },
+      ]);
+    });
+
+    it("should be able to complete a transaction without going through checkoutCart", async () => {
```

Comment:
Hmm come to think of it ah, it'll be better if we restrict it such that the state must be "PENDING_CONFIRMATION" when `completeCheckout` is called. This ensures `checkoutCart`, with all its validations, have already been carried out